compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -144,6 +150,23 @@ private JMethod resolve(JMethodRequest jMethodRequest, List<JMethod> jMethods) {
         return jMethod;
     }
 
+    private Optional<JMethod> findCovariantReturnTypeMethod(List<JMethod> jMethods) {
+
+        for (JMethod ithMethod : jMethods) {
+            for (JMethod kthMethod : jMethods) {
```

Comment:
here we can reduce the number of iterations by half by changing `i=1..N x k=1..N` to `i=1..N x k=i..N`

Reply:
Yep, updated the logic.